File: kotlin/Problem026.kt
import java.math.BigDecimal

fun main(args: Array<String>) {
    EulerSolution.execute(Problem026())
}

/**
 * Not done!
 */
class Problem026 : EulerSolution {
    override fun run() : Long {
        val one = BigDecimal.ONE.setScale(2000)

        var longest : Int

        for (d in 1..10) {
            val big = one / BigDecimal.valueOf(d.toDouble()).setScale(2000)
            val str = big.toString().substringAfter('.')

            var max = 0

            for (i in 1..5) {
                val sub = str.substring(0, i)
                println(sub + sub + sub)
                if (str.startsWith(sub + sub + sub)) {
                    println("--- YES")
                    max = i
                }
            }

            println("$d = $max")
        }

        return 0
    }
}

File: c/problem026.c
#include "euler_solution.c"

long long run() {
    int result;
    int length = 0;
    int remainders[1000] = {0};

    for (int i = 1000; i > 1; i--) {
        if (length >= i)
            break;

        // set array to 0
        for (int j = 0; j < i; j++) {
            remainders[j] = 0;
        }

        int value = 1;
        int current = 0;
        while (value > 0 && remainders[value] == 0) {
            remainders[value] = current;
            value *= 10;
            value %= i;
            current++;
        }

        if (current - remainders[value] > length) {
            result = i;
            length = current - remainders[value];
        }
    }

    return result;
}
